### DISCUSSION TITLE
Recommended way to initialize first user

### DISCUSSION THREAD

**Discussion by megafinz** [keystonejs/keystone#7880] [Questions] [answered]
I (and, as I see by code snippets in issues and disccussions, along with many others :)) was following the Wes Bos' [course](https://advancedreact.com) that among everyting demonstrates how to organize authorization by utilizing separate `Role` entities that contain a list of permissions. These `Role` entities are then associated with `User` and used to govern the access to other entities.

What unfortunately is not covered in this course and what I'm having trouble to figure out is: how do I assign a role to the initial `User` that is created during initialization process? 
- I was able to create the role itself on the first run using the `config.db.onConnect` hook.
- I tried displaying the role in the registration UI using the `initFirstItem.fields` prop, but unfortunately authorization rules prevent me from seeing/setting it.
- I've seen the `initFirstItem.itemData` prop but looks like it is used to set props directly on the new `User` entity and there is no way to query a DB to find the role ID to set. I understand that I can introduce a `boolean` prop directly on the `User` entity (e.g. `User.isSuperUser`), but that is not very clean and I'd like to avoid doing that.
- My current solution is to adjust `User.roleId` for the initial `User` directly on the database after initial registration, which is manageable, but not very convenient.

Is there any recommended way to handle this?

Thanks.

**Comment by flexdinesh** [comment, marked as answer]
You can create first item with relational values. We did a workshop for Prisma Day 2021 and the init logic created a user with superuser role. You can find the code for it [here](https://github.com/keystonejs/prisma-day-2021-workshop/blob/fd191293619f93a4c70b8f4fcc98348dfc61ce1d/keystone.ts#L24). 

Hope it helps.

**Reply by megafinz** [reply]
Wow sir this works like a charm. Thank you very much! Wish documentation was a little bit clearer on this topic because it's not very obvious you can provide relational data like that.
